clientv3/client.go

```diff
@@ -327,9 +328,14 @@ func (c *Client) getToken(ctx context.Context) error {
 }
 
 func (c *Client) dial(ep string, dopts ...grpc.DialOption) (*grpc.ClientConn, error) {
-	opts, err := c.dialSetupOpts(ep, dopts...)
+	// We pass a target to DialContext of the form: endpoint://<clusterName>/<host-part> that
+	// does not include scheme (http/https/unix/unixs) or path parts.
+	_, host, _ := endpoint.ParseEndpoint(ep)
+	target := c.resolver.Target(host)
```

Comment:
This is the main fix in this PR-- moving the `resolver.Target` call to here guarentees it's applied for all codepaths.